TestChildrenExpiration lockup

## Expected Behavior
Tests should always finish, red or green.

## Current Behavior
(truncated to fit into 64K GH limit)
```
panic: test timed out after 10m0s
	running tests:
		TestChildrenExpiration (9m56s)

goroutine 843 [running]:
testing.(*M).startAlarm.func1()
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:2484 +0x605
created by time.goFunc
	/opt/hostedtoolcache/go/1.24.13/x64/src/time/sleep.go:215 +0x45

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc0002a3340, {0x1333170, 0x16}, 0x14184f8)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1859 +0x91e
testing.runTests.func1(0xc0002a3340)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:2279 +0x86
testing.tRunner(0xc0002a3340, 0xc00024bae0)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1792 +0x226
testing.runTests(0xc000282288, {0x1c299a0, 0x22, 0x22}, {0x1?, 0x0?, 0x1c37cc0?})
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:2277 +0x96d
testing.(*M).Run(0xc0002be140)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:2142 +0xeeb
main.main()
	_testmain.go:113 +0x165

goroutine 19 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc000250b60)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 1
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 20 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc000250b60)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 1
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 335 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00015ea90)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 316
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 122 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x5)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 191 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000123bc0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 161
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 117 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 750 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572600)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 121 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x4)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 119 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x2)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 124 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x7)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 336 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00015ea90)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 316
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 127 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0xa)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 125 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x8)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 120 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x3)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 118 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x1)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 331 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00015e820)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 316
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 787 [sync.RWMutex.Lock, 9 minutes]:
sync.runtime_SemacquireRWMutex(0xc000304cfc?, 0x1?, 0x1002d90?)
	/opt/hostedtoolcache/go/1.24.13/x64/src/runtime/sema.go:105 +0x25
sync.(*RWMutex).Lock(0xc000304ce8)
	/opt/hostedtoolcache/go/1.24.13/x64/src/sync/rwmutex.go:155 +0x89
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setBlockExecErr(0xc000304cb0, {0x1546240, 0x1c12290})
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:106 +0x65
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Close(0xc000304cb0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:76 +0xd0
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.TestChildrenExpiration.func1()
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/gc_test.go:54 +0x2f
testing.(*common).Cleanup.func1()
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1211 +0x170
testing.(*common).runCleanup(0xc0003af6c0, 0x0)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1445 +0x2b4
testing.tRunner.func2()
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1786 +0x4d
testing.tRunner(0xc0003af6c0, 0x14184f8)
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1798 +0x25f
created by testing.(*T).Run in goroutine 1
	/opt/hostedtoolcache/go/1.24.13/x64/src/testing/testing.go:1851 +0x8f3

goroutine 320 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0003c24e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 316
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 11 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc000204a80)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 213
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 77 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00039f110)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 76
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 193 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000123ce0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 161
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 123 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x6)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 194 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc00030a3f0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 161
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 126 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0x9)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 444 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0000996e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 439
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 321 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0003c2660)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 316
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 128 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0xb)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 129 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0xc)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 130 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushWorker(0xc0002dc240, 0xd)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:132 +0x1af
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:36 +0x67

goroutine 137 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).flushScheduler(0xc0002dc240)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:49 +0x1ef
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/writecache.(*cache).runFlushLoop in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/writecache/flush.go:40 +0x185

goroutine 138 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0004770e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 139 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0004770e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 116
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 186 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000123980)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 161
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 210 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000098120)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 161
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 211 [select]:
goroutine 223 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000098540)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 213
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 615 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc0004e2460)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 542
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 753 [sync.RWMutex.RLock, 9 minutes]:
sync.runtime_SemacquireRWMutexR(0xc000304cf8?, 0x1?, 0x1017465?)
	/opt/hostedtoolcache/go/1.24.13/x64/src/runtime/sema.go:100 +0x25
sync.(*RWMutex).RLock(0xc000304ce8)
	/opt/hostedtoolcache/go/1.24.13/x64/src/sync/rwmutex.go:74 +0x5b
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Get(0xc000304cb0, {{0x3c, 0xa9, 0x7, 0xc9, 0x47, 0xd8, 0x5c, 0x4e, 0xb2, ...}, ...})
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/get.go:34 +0x211
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).processAddrDelete(0xc000304cb0, {{0x3c, 0xa9, 0x7, 0xc9, 0x47, 0xd8, 0x5c, 0x4e, 0xb2, ...}, ...}, ...)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/inhume.go:99 +0x1065
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).processExpiredObjects(0xc000304cb0, {0xc00050ee40, 0x1, 0x1336669?})
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/inhume.go:269 +0x4c6
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*Shard).collectExpiredObjects(0xc000035240)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:238 +0xb2e
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*Shard).removeGarbage(0xc000035240)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:149 +0x125
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000572960)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:120 +0x116
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 752 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572840)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 802 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572960)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 497 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000476c00)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 439
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 498 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000476c00)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 439
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 546 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0001226c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 547 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0001226c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 548 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0001228a0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 549 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0001228a0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 550 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000123c20)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 551 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000123c20)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 552 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0003c25a0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 553 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0003c25a0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 554 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0003c2a20)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 555 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0003c2a20)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 529
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 572 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0003c3d40)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 573 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0003c3d40)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 574 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0003c3ec0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 575 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0003c3ec0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 576 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0005720c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 577 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0005720c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 578 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000572240)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 579 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572240)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 580 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0005723c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 581 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0005723c0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 571
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 582 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00030c410)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 583 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00030c410)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 584 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00030c4e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 585 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00030c4e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 586 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00030c5b0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 587 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00030c5b0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 588 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00030cdd0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 589 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00030cdd0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 590 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc0003b2000)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 591 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc0003b2000)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 571
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 789 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc0003b28f0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 787
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 636 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000477e60)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 633
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 751 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000572840)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 749 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000572600)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 635 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc0003b24e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 633
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 647 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00015ef70)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 658
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 618 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc0000f3520)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 617
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 619 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc0000f3520)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 617
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 620 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc0000f3ad0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 617
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 621 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc0000f3ad0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 617
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 622 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000573440)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 617
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 623 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000573440)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 617
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 624 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000573560)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 617
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 625 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000573560)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 617
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 642 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc0004e3570)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 617
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 648 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000593260)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 658
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 764 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000573a40)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 754
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 649 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000593260)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 658
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 650 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000593560)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 658
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 651 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000593560)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 658
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 652 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc000279030)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 658
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 653 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00039e4e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 654 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00039e4e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 655 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00039eb60)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 656 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00039eb60)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 657 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).purgeStaleWorkers(0xc00039f1e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:253 +0x20c
created by github.com/panjf2000/ants/v2.(*poolCommon).goPurge in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:317 +0x185

goroutine 674 [select]:
github.com/panjf2000/ants/v2.(*poolCommon).ticktock(0xc00039f1e0)
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:296 +0x22c
created by github.com/panjf2000/ants/v2.(*poolCommon).goTicktock in goroutine 627
	/home/runner/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/ants.go:323 +0x191

goroutine 675 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc0002410e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 676 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc0002410e0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 677 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000241200)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 678 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000241200)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 679 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000241320)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 680 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000241320)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 681 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc000600fc0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 627
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 804 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572a80)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 637 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000477e60)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 633
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b

goroutine 638 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc0004e3880)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 633
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 716 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc000600690)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 707
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 771 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).setModeLoop(0xc0003d1340)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/engine.go:77 +0x1e5
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine.(*StorageEngine).Init in goroutine 754
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/engine/control.go:62 +0x7eb

goroutine 805 [select]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).tickRemover(0xc000572ba0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:115 +0x205
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:81 +0xa5

goroutine 806 [select, 9 minutes]:
github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).listenEvents(0xc000572ba0)
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:89 +0x185
created by github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard.(*gc).init in goroutine 787
	/home/runner/work/neofs-node/neofs-node/pkg/local_object_storage/shard/gc.go:82 +0x10b
```

## Possible Solution
Fix something.

## Steps to Reproduce (for bugs)
Run the test.

## Context
https://github.com/nspcc-dev/neofs-node/actions/runs/22180665840/job/64140817915?pr=3833

## Regression
Nope.

## Your Environment
* Version used: 0.51.1
* Server setup and configuration:
* Operating System and version (`uname -a`):

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ Changelog for NeoFS Node
 
 ### Fixed
 - Resending the header after chunks have already been sent in object service `Get` handler (#3833)
+- GC deadlock on local object storage shutdown (#3837)
 
 ### Changed
 

--- a/pkg/local_object_storage/engine/get.go
+++ b/pkg/local_object_storage/engine/get.go
@@ -37,7 +37,11 @@ func (e *StorageEngine) Get(addr oid.Address) (*object.Object, error) {
 	if e.blockErr != nil {
 		return nil, e.blockErr
 	}
+	return e.getInt(addr)
+}
 
+// getInt is an unlocked version of Get.
+func (e *StorageEngine) getInt(addr oid.Address) (*object.Object, error) {
 	var (
 		err error
 		obj *object.Object

--- a/pkg/local_object_storage/engine/inhume.go
+++ b/pkg/local_object_storage/engine/inhume.go
@@ -96,7 +96,7 @@ func (e *StorageEngine) processAddrDelete(addr oid.Address, deleteFunc func(*sha
 			linkAddr.SetContainer(addr.Container())
 			linkAddr.SetObject(linkID)
 
-			linkObj, err := e.Get(linkAddr)
+			linkObj, err := e.getInt(linkAddr)
 			if err != nil {
 				e.log.Debug("inhuming root object but no link object is found",
 					zap.Stringer("linkAddr", linkAddr),

